model/aberrations.js

```diff
@@ -64,66 +71,198 @@ exports.upsertAber = function(data, callback){
+	acids = data.protein_seq_change.split(/\d+/);
+	locus = 0; 	
+	if (acids[0].length < data.protein_seq_change.length)
+	    locus = parseInt(data.protein_seq_change.substr(acids[0].length));
+	if (acids.length == 1)
+	    acids[1] = '?';
+    }
 
+    
     var aberInsertQuery = abers.insert(
-	abers.gene.value(data.gene),
-	abers.transcript.value(data.transcript),
-	abers.mut_class.value(data.mut_class),
-	abers.mut_type.value(data.mut_type),
-	abers.protein_seq_change.value(data.protein_seq_change),
-	abers.aber_user_id.value(data.user_id)).returning(abers.u_id);
-
-    Database.execute(aberInsertQuery, function(err, subresult) {
-	handleErr(err, subresult, aberInsertQuery);
-	var aber_u_id = subresult.rows[0].u_id;
-	data.aber_id = aber_u_id;
-	console.log(data);
-	exports.upsertSourceAnno(data, callback);
-    });
+	abers.gene_id.value(data.gene),
+	abers.mutation_class.value(data.mut_class),
+	abers.mutation_type.value(data.mut_type),
+	abers.locus.value(locus),
+	abers.original_amino_acid.value(acids[0]),
+	abers.new_amino_acid.value(acids[1]),
+	abers.last_edited.value(now),
+	abers.created_on.value(now)).returning(abers.id);
+
+	DjangoDatabase.execute(aberInsertQuery, function(err, subresult) {
+	    handleErr(err, subresult, aberInsertQuery);
+	    if (!err) {
+		var aber_u_id = subresult.rows[0].id;
+		data.aber_id = aber_u_id;
+		exports.upsertSourceAnno(data, callback);
+	    }
+	});
 }
 
-// note: requires an aber_id field to identify which aberration this source attaches to
-exports.upsertSourceAnno = function(data, callback) {
-    var sources = Schemas.aber_sources;
+// callback supplies user data
+function getUserInfo(user, callback) {
 
-    // only column in the definition of the source should survive
-    var keys = Object.keys(data);
-    var columns = Schemas.getColumnNames(sources);
-    keys.forEach(function(key) {if (!columns.some(function (c) {return key == c;})) delete data[key];});
-    // check if a record exists
-    var updateQuery = sources.update(data).
-	where(sources.aber_id.equals(data.aber_id),
-	      sources.user_id.equals(data.user_id)).
-	returning(sources.asa_u_id);
-    Database.execute(updateQuery, function(err, result) {
+    DjangoDatabase.execute(
+	Schemas.users.where({'email': user.email}),
+	function (err, user_subresult) {
+	    if(err) 
+		callback(err, null);
+	    else if (user_subresult.rows.length == 1) 
+		callback(null, user_subresult.rows[0]);
+	    else if (user_subresult.rows.length == 0) {
+		first_last_names = user.name.split(" ");
+		if (first_last_names.length == 1) 
+		    first_last_names[1] = '';
+
+		// todo: pass up error with error requesting login 
+		DjangoDatabase.execute(
+		    Schemas.users.insert({'first_name': first_last_names[0],
+				  'last_name': first_last_names[1],
+				  'email': user.email,
+				  'username': user.email.split('@')[0],
+				  'is_staff': false,
+				  'is_superuser': false,
+				  'is_active': false,
+				  'date_joined': getTime("day"),
```

Comment:
in order to serialize the date, postgresql wants a string in date format, and the default string conversion for the date wasn't compatible with postgresql for some reason